clippy_lints/src/unnecessary_wraps.rs

```diff
@@ -140,7 +140,7 @@ impl<'tcx> LateLintPass<'tcx> for UnnecessaryWraps {
                     UNNECESSARY_WRAPS,
                     fn_decl.output.span(),
                     "unneeded wrapped unit return type",
-                    format!("remove the `-> {}<()>`", return_type_label).as_str(),
+                    format!("remove the `-> {}<[...]>`", return_type_label).as_str(),
```

Comment:
I'm not too sure how to write this to be consistent with both Option and Result.

Reply:
You can just say "remove this return type". No need to repeat the code that is shown and underlined.

The current version of the lint says

```text
...and change the returning expressions
```

For the new cases added here, could you just change that message to

```text
...and remove instances of `Some(())`
```

I don't think it's really necessary to detect whether the line needs to be removed. Unless I'm missing something?

If you go with a more "detailed" approach, just make sure there are more test cases to go with it.

Reply:
Thank you for you comment! :-)
Just to be certain to understand, are you suggesting that we do not provide suggestions for the body of the function? Is that what you call the detailed approach?

Reply:
I was not very clear. I meant to use `diag.multipart_suggestion(message, suggs, ..)` just like the existing code, but just change the message for the "unit case" that you are adding. By "detailed" approach, I am referring to the new approach for linting the body that you have been discussing above.